sydent/config/__init__.py

```diff
@@ -200,55 +201,58 @@ def _parse_config(self, cfg: ConfigParser) -> bool:
 
         return needs_saving
 
-    def parse_from_config_parser(self, cfg: ConfigParser) -> bool:
+    def _parse_from_config_parser(self, cfg: ConfigParser) -> bool:
         """
         Parse the configuration from a ConfigParser object
 
         :param cfg: the configuration to be parsed
 
-        :return: whether or not cfg has been altered. This method CAN
-            return True, but it *shouldn't* as this leads to altering the
-            config file.
+        :return: whether or not the config file needs updating. This method CAN
+            return True, but it *shouldn't*. Instead a ConfigError exception
+            should be raised. This is left in for the soon to be deprecated way
+            of generating config files.
         """
         return self._parse_config(cfg)
 
-    def parse_config_file(self, config_file: str) -> None:
+    def parse_config_file(
+        self, config_file: str, skip_logging_setup: bool = False
+    ) -> None:
         """
         Parse the given config from a filepath, populating missing items and
         sections. NOTE: this method also sets up logging.
 
         :param config_file: the file to be parsed
         """
         # If the config file doesn't exist, prepopulate the config object
-        # with the defaults, in the right section.
-        #
-        # Otherwise, we have to put the defaults in the DEFAULT section,
-        # to ensure that they don't override anyone's settings which are
-        # in their config file in the default section (which is likely,
-        # because sydent used to be braindead).
-        use_defaults = not os.path.exists(config_file)
+        # with the defaults, in the DEFAULT section.
+        new_config_file = not os.path.exists(config_file)
```

Comment:
this logic is still backwards (https://github.com/matrix-org/sydent/issues/314), but I guess that doesn't matter too much since it's going away in #402.

Reply:
The logic is backwards, but I updated the comment to say what the code actually does (which is populate the DEFAULT section on the first run). But yes, it goes away in #402